### PR TITLE
Add Bill of Materials

### PR DESCRIPTION
**Pull request by snicoll**
This commit adds a BOM with the dependencies that this project should
manage. To make sure that the BOM is valid, the project has been
restructured to use the BOM internally.

As a result, the root POM is only used to invoke the reactor and does
not need to be published anymore. A new parent, inheriting from the bom
provides the required build configuration and external dependency
management.

### CODE REVIEW

**Comment by snicoll** [comment]
Great question. They won't and we use the flatten-maven-plugin anyway so all the unnecessary stuff is reduced. If you build that branch, here is what the published bom looks like:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.wavefront</groupId>
  <artifactId>wavefront-spring-boot-bom</artifactId>
  <version>2.0.0-RC1-SNAPSHOT</version>
  <packaging>pom</packaging>
  <name>Wavefront Spring Boot Bill of Materials</name>
  <description>Wavefront by VMware integration for Spring Boot</description>
  <url>https://www.wavefront.com/wavefront-spring-boot-bom</url>
  <organization>
    <name>VMware, Inc.</name>
    <url>https://www.vmware.com</url>
  </organization>
  <licenses>
    <license>
      <name>Apache License, Version 2.0</name>
      <url>https://www.apache.org/licenses/LICENSE-2.0</url>
    </license>
  </licenses>
  <developers>
    <developer>
      <name>Clement Pang</name>
      <email>clement@wavefront.com</email>
      <organization>Wavefront</organization>
      <organizationUrl>https://www.wavefront.com</organizationUrl>
    </developer>
  </developers>
  <scm>
    <connection>scm:git:git@github.com:wavefrontHQ/wavefront-spring-boot.git/wavefront-spring-boot-bom</connection>
    <developerConnection>scm:git:git@github.com:wavefrontHQ/wavefront-spring-boot.git/wavefront-spring-boot-bom</developerConnection>
    <url>git@github.com:wavefrontHQ/wavefront-spring-boot.git/wavefront-spring-boot-bom</url>
  </scm>
  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>com.wavefront</groupId>
        <artifactId>wavefront-spring-boot</artifactId>
        <version>${project.version}</version>
      </dependency>
      <dependency>
        <groupId>com.wavefront</groupId>
        <artifactId>wavefront-spring-boot-starter</artifactId>
        <version>${project.version}</version>
      </dependency>
      <dependency>
        <groupId>com.wavefront</groupId>
        <artifactId>wavefront-opentracing-sdk-java</artifactId>
        <version>1.16</version>
      </dependency>
      <dependency>
        <groupId>com.wavefront</groupId>
        <artifactId>wavefront-sdk-java</artifactId>
        <version>2.3</version>
      </dependency>
    </dependencies>
  </dependencyManagement>
</project>

```

File: wavefront-spring-boot-bom/pom.xml [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                  <goal>sign</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-javadoc-plugin</artifactId>
+            <configuration>
+              <source>${java.version}</source>
```
Comment (ajayj89) [inline wavefront-spring-boot-bom/pom.xml:114]: this property is defined in child `wavefront-spring-boot-parent` module, should we move the property to this module for clarity?

File: pom.xml [hunk 2]
```diff
@@ -40,266 +40,30 @@
...
-          <version>${spring-boot.version}</version>
-        </plugin>
-      </plugins>
-    </pluginManagement>
-
     <plugins>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-compiler-plugin</artifactId>
+        <artifactId>maven-deploy-plugin</artifactId>
```
Comment (ajayj89) [inline pom.xml:57]: i don't know much about bom, but why do we need deploy plugin here and not just including this in the child `wavefront-spring-boot-parent` module, similar to release / javadoc / compile and other maven plugins ?

Comment (snicoll) [reply pom.xml:57]: This tells that the root bom shouldn't be deployed at all. We don't need it as the flatten maven plugin takes care of doing the work on the bom. I've pasted an example as a comment.

**Comment by panghy** [comment]
LGTM then

**Pull request merged**